packages/block-editor/src/store/reducer.js

```diff
@@ -1829,6 +1829,8 @@ export function highlightedBlock( state, action ) {
 export function lastBlockInserted( state = {}, action ) {
 	switch ( action.type ) {
 		case 'INSERT_BLOCKS':
+		case 'REPLACE_BLOCKS':
+		case 'REPLACE_INNER_BLOCKS':
```

Comment:
Does this make sense at all? 

Both `REPLACE_` calls can be made with multiple blocks. Therefore how can we decide which block's clientId to return as the "last inserted"? Is it the first or the last or one of the others?

Or should we make the selector experimental and make it plural. So the user can expect one/many blocks to be returned? Honestly I should have made it unstable or experimental to start with 😟

Reply:
Can INSERT_BLOCKS also be called with multiple blocks?

Could we return an array of all the inserted blocks?

Reply:
This seems to be useful when we insert one block and it always returns the 1st one in the list. It was added by #29546 refactored by #30123.

Doing a code search it seems it remained used exclusively by the `getLastInsertedBlockClientId` selector required by some blocks to know they've been inserted so some action can be done.

I think adding replace block actions to setting this selector makes sense, but only when we want to replace ONE block.

Reply:
@draganescu An alternative might be https://github.com/WordPress/gutenberg/pull/46885.

Reply:
In https://github.com/WordPress/gutenberg/pull/46885 we decided to create a new selector which returns an array of clientIDs.